cpp/unit/unit.h

```diff
@@ -70,6 +71,41 @@ class Unit : public log::LogSource {
 	 */
 	graphic_set *graphics;
 
+	/**
+	 * constructs a new location for this unit replacing any
+	 * existing location
+	 *
+	 * uses same args as the location constructor
+	 * except the first which will filled automatically
+	 */
+	template<class T, typename ... Arg>
```

Comment:
Well I guess it's needed because `T` is templated. BTW you should add a `static_assert` that checks that `T` is a subclass of `TerrainObject`